main.c

```diff
@@ -1177,10 +1181,18 @@ static void loadUsbDModule(void)
 //---------------------------------------------------------------------------
 static void loadUsbModules(void)
 {
+    int ret;
+
     loadUsbDModule();
-    if (have_usbd && !have_usb_mass && (USB_mass_loaded = loadExternalModule(setting->usbmass_file, &usb_mass_irx, size_usb_mass_irx))) {
+    if (have_usbd && !have_usb_mass && (USB_mass_loaded = loadExternalModule(setting->usbmass_file, NULL, 0))) {
         delay(3);
         have_usb_mass = 1;
+    } else if (have_usbd && !have_usb_mass) {
+        SifExecModuleBuffer(bdm_irx, size_bdm_irx, 0, NULL, &ret);
+        SifExecModuleBuffer(bdmfs_fatfs_irx, size_bdmfs_fatfs_irx, 0, NULL, &ret);
+        SifExecModuleBuffer(usbmass_bd_irx, size_usbmass_bd_irx, 0, NULL, &ret);
+        USB_mass_loaded = 1;
```

Comment:
No `delay(3)` here?
If it's needed it should probably be here.

Reply:
I remember that on deckard we should wait before usb is recognized

Reply:
Hmm… I'll add the delay back in then.

Reply:
If you can figure out why a delay is required and engineer it out of existence, that would be great.